components/feature/downloads/src/test/java/mozilla/components/feature/downloads/DownloadMiddlewareTest.kt

```diff
@@ -186,6 +214,7 @@ class DownloadMiddlewareTest {
         assertTrue(store.state.downloads.isEmpty())
 
         store.dispatch(DownloadAction.RestoreDownloadsStateAction).joinBlocking()
+        dispatcher.advanceUntilIdle()
 
```

Comment:
@Amejia481 I've just noticed we need both a `dispatcher.advanceUntilIdle` and also a `store.waitUnitldle` here to fix the intermittent test failure. The first one to make sure we've launched the coroutine in `restoreDownload` that dispatches the action. The second one to make sure the action has been processed by the store. We should have landed both PRs :).

Right now if we simulate a slow processing of the action the test would still fail.

Reply:
Thanks, I will put a pr to fix!

Reply:
P
T
O

Reply:
![](https://media.giphy.com/media/4bpK2k0Yru5Us/giphy.gif)